setup1-contributor/src/reliability/cpu.rs

```diff
@@ -24,3 +57,25 @@ where
     write_half.send(response).await.map_err(|e| anyhow!("{:?}", e))?;
     Ok(())
 }
+
+#[cfg(test)]
+mod test {
+    use crate::reliability::cpu::calculate_powers_of_tau;
+    use std::time::Instant;
+    use zexe_algebra::Bls12_377;
+
+    const DATA_LENGTH: usize = 64;
+    const OUTPUT_LENGTH: usize = 790192;
+
+    #[test]
+    fn test_calculate_powers_of_tau() {
+        let data: Vec<u8> = vec![0; DATA_LENGTH];
+        let now = Instant::now();
+        let output = calculate_powers_of_tau::<Bls12_377>(data);
+        let elapsed = now.elapsed();
+        println!("Proof of Work took: {:.2?}", elapsed);
+        assert_eq!(output.len(), OUTPUT_LENGTH);
+        let expected = [12, 53, 232, 21, 121, 10, 125, 129];
```

Comment:
am I missing something or shouldn't 0 power n still be 0?

Reply:
The input data is passed to construct the RNG, which is used to generate the key pair. Then we can consider the input data as a seed. Using this seed we obtain non-zero key pair, and the computation of the powers of tau works as usual.